oxts-sdk-gal-cpp/include/oxts/gal-cpp/gad.hpp

```diff
@@ -572,6 +572,113 @@ namespace OxTS
 		std::vector<double> GetAidingLeverArmVar() const;
 	};
 
+	/**
+	 * Generic Aiding speed.
+	 */
+	class GadSpeed : public Gad
+	{
+	private:
+
+	public:
+		explicit GadSpeed(uint8_t stream_id);
+
+		/**
+		 * Set the forward speed aiding estimate.
+		 *
+		 * @param speed Forward speed estimate in the vehicle frame (m/s).
+		 */
+		void SetSpeedFwMs(double speed);
+		/**
+		 * Set the backward speed aiding estimate.
+		 *
+		 * @param speed Backward speed estimate in the vehicle frame (m/s).
+		 */
+		void SetSpeedBwMs(double speed);
+		/**
+		 * Set an unsigned speed aiding estimate.
+		 *
+		 * @param speed Unsigned speed estimate in the vehicle frame (m/s).
+		 */
+		void SetSpeedUnMs(double speed);
+		/**
+		 * Set the forward pulsed speed aiding estimate.
+		 *
+		 * @param frequency Pulse frequency in the forward vehicle direction (counts/s).
+		 * @param scale_factor Scale factor converting counts to m, e.g. 0.01 for 100 counts/m. (m)
+		 * @param(optional) period Time over which the speed measurement is taken (if known) (s).
+		 */
+		void SetSpeedFwPulsed(double frequency, double scale_factor, double period=0.0);
```

Comment:
If a user is sending pulses do they not want to be able to send a raw pulse count? A wheelspeed sensor gives pulses which would then have to be converted to a frequency by dividing by the period, which means period isn't really optional. Would recommend period is made non-optional and frequency is changed to pulses - we can do the conversion in function if necessary.

Reply:
Also zero period does not play nicely in the nav engine - protection needed.

Reply:
For current build, will leave as-is to avoid issue with variance error propagation. Period default will change.